embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/ndk/EmbraceNdkServiceTest.kt

```diff
@@ -541,11 +553,73 @@ internal class EmbraceNdkServiceTest {
     @Test
     fun `test initialization does not does not install signals and create directories if loadEmbraceNative is false`() {
         enableNdk(true)
-        every { sharedObjectLoader.loadEmbraceNative() } returns false
+        every { sharedObjectLoader.loadEmbraceNative() } returns true
+        initializeService()
+        verify(exactly = 0) { embraceNdkService["installSignals"]() }
+        verify(exactly = 0) { embraceNdkService["createCrashReportDirectory"]() }
+    }
+
+    @Test
+    fun `initialization happens synchronously by default`() {
+        enableNdk(true)
+        blockableExecutorService.blockingMode = true
+        initializeService()
+        assertNativeSignalHandlerInstalled()
```

Comment:
Verifying the method call on the delegate because there's no good way to inspect the side effects otherwise. Ah well.